python/graphstorm/config/argument.py

```diff
@@ -1302,21 +1335,38 @@ def imbalance_class_weights(self):
 
             Customer should provide the weight in following format 0.1,0.2,0.3,0.1
         """
-        if hasattr(self, "_imbalance_class_weights"):
+
+        def check_imbalance_class_weights(imbalance_class_weights, num_classes):
             try:
-                weights = self._imbalance_class_weights.split(",")
+                weights = imbalance_class_weights.split(",")
                 weights = [float(w) for w in weights]
             except Exception: # pylint: disable=broad-except
-                assert False, \
-                    "The rescaling weights should in following format 0.1,0.2,0.3,0.1"
+                raise RuntimeError("The weights should in following format 0.1,0.2,0.3,0.1")
             for w in weights:
                 assert w > 0., "Each weight should be larger than 0."
-            assert len(weights) == self.num_classes, \
+            assert len(weights) == num_classes, \
                 "Each class must have an assigned weight"
-
             return th.tensor(weights)
 
-        return None
+        if hasattr(self, "_num_classes") and isinstance(self.num_classes, dict):
+            if hasattr(self, "_imbalance_class_weights"):
+                assert isinstance(self._imbalance_class_weights, dict), \
+                    print('The imbalance_class_weights should be dictionary')
+                num_classes = self.num_classes
+                imbalance_class_weights = self._imbalance_class_weights
+                ntype_weights = {}
+                for ntype in num_classes:
+                    ntype_weights[ntype] = check_imbalance_class_weights(
+                        imbalance_class_weights[ntype],
```

Comment:
What if some node types have imbalance class weights and some node types do not?

Reply:
solved.